src/components/main-view/main-view.jsx

```diff
@@ -1,79 +1,83 @@
-            genre: [{
-                name: "Comedy",
-                description: "Comedy is a genre of film in which the main emphasis is on humor. These films are designed to make the audience laugh through amusement and most often work by exaggerating characteristics for humorous effect."
-            }],
-            director: [{
-                name: "Lake Bell",
-                bio: "Lake Siegel Bell is an American actress, screenwriter, and director. She wrote and directed the short film Worst Enemy, which debuted at the Sundance Film Festival in 2012, followed by her 2013 feature film directing debut In a World..., in which she also starred.",
-                birth: "1979"
-            }]
-        },
-        {
-            id: 3,
-            title: "Marie Antoinette",
-            description: "The retelling of France's iconic but ill-fated queen, Marie Antoinette. From her betrothal and marriage to Louis XVI at 14 to her reign as queen at 19 and to the end of her reign as queen, and ultimately the fall of Versailles.",
-            image: "https://m.media-amazon.com/images/M/MV5BNDViNjYxNWYtMWRiOS00NmMyLTgwNDMtNTg3MWYwNjAxNWM4XkEyXkFqcGdeQXVyMjUzOTY1NTc@._V1_FMjpg_UX1000_.jpg",
-            genre: [{
-                name: "Drama",
-                description: "In film and television, drama is a category or genre of narrative fiction (or semi-fiction) intended to be more serious than humorous in tone. A primary element in a drama is the occurrence of conflict—emotional, social, or otherwise—and its resolution in the course of the storyline."
-            }],
-            director: [{
-                name: "Sofia Coppola",
-                bio: "Sofia Coppola is an American director and actress. She acted in The Godfather trilogy and Star Wars Episode I: The Phantom Menace. She directed Lost in Translation, Marie Antoinette, The Virgin Suicides, The Beguiled, On the Rocks, Somewhere and The Bling Ring. She is the daughter of Francis Ford Coppola.",
-                birth: "1971"
-            }]
-        },
-    ]);
 
-const [selectedMovie, setSelectedMovie] = useState(null);
+    const storedUser = JSON.parse(localStorage.getItem("user"));
+
+    const storedToken = localStorage.getItem("token");
+
+    const [user, setUser] = useState(storedUser ? storedUser : null);
+
+    const [token, setToken] = useState(storedToken ? storedToken : null);
+
+    const [movies, setMovies] = useState([]);
+
+    const [selectedMovie, setSelectedMovie] = useState(null);
+
+
+useEffect(() => {
+    if (!token) {
+        return;
+    }
+
+    fetch("https://jens-movie-api.herokuapp.com/movies", {
+        headers: { Authorization: `Bearer ${token}` }
+    })
+    .then((response) => response.json())
+    .then((data) => {
+        const movies = data.docs.map((doc) => {
+            return {
+                id: doc.key,
```

Comment:
where does key come from? do you mean `_id`?

Reply:
Buddy, you need to provide more context for me, because I don't KNOW if I meant "_id" or "id", because this is literally the way the Exercise showed me how to write this.

Reply:
you dont have a field called key based on this code
https://github.com/jenscotch/movie_api/blob/main/models.js
<img width="534" alt="image" src="https://github.com/jenscotch/myFlix-client/assets/13005417/9ebb6c7d-e205-40ba-9428-76a51411b534">

but mongodb creates a field called `_id` as an identifier to uniquely identify all documents in a collection (look at achievement 2.7 and https://www.mongodb.com/docs/manual/core/document/). Thats why it should be `_id`